crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -44,6 +45,7 @@ pub struct Solution {
     score: SolverScore,
     weth: eth::WethAddress,
     gas: Option<eth::Gas>,
+    fee_handler: FeeHandler,
```

Comment:
fee_handler is not really solution specific, but rather solver specific. Can we read this field transitively from the `solver` field instead?